File: Examples/ObjC/Sources/ContentCards-Custom-UI/Readme.m
#import "AppDelegate.h"
#import "ReadmeAction.h"
#import "ReadmeViewController.h"

#pragma mark - Internal

void printCurrentContentCards(ReadmeViewController *viewController) {
  [(AppDelegate *)
          UIApplication.sharedApplication.delegate printCurrentContentCards];
}

void refreshContentCards(ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication.delegate refreshContentCards];
}

void subscribeToContentCardsUpdates(ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication
          .delegate subscribeToContentCardsUpdates];
}

void cancelContentCardsUpdatesSubscription(
    ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication
          .delegate cancelContentCardsUpdatesSubscription];
}

void presentContentCardsInfoViewController(
    ReadmeViewController *viewController) {
  [(AppDelegate *)UIApplication.sharedApplication
          .delegate presentContentCardsInfoViewController];
}

#pragma mark - Readme

NSString *const readme = @"This sample demonstrates how to implement your own "
                         @"custom Content Cards UI:\n"
                         @"\n"
                         @"- AppDelegate.{h,m}:\n"
                         @"  - Content cards API usage\n"
                         @"  - Content cards custom UI presentation\n"
                         @"- CardsInfoViewController.{h,m}\n"
                         @"  - UIViewController subclass presenting the "
                         @"content cards data in a table view";

NSInteger const actionsCount = 5;

ReadmeAction const actions[] = {
    {@"Print current Content Cards", @"",
     ^(ReadmeViewController *_Nonnull viewController){
         printCurrentContentCards(viewController);
}
}
, {@"Refresh Content Cards", @"",
   ^(ReadmeViewController *_Nonnull viewController){
       refreshContentCards(viewController);
}
}
, {@"Subscribe to Content Cards updates", @"",
   ^(ReadmeViewController *_Nonnull viewController){
       subscribeToContentCardsUpdates(viewController);
}
}
, {@"Cancel Content Cards updates subscription", @"",
   ^(ReadmeViewController *_Nonnull viewController){
       cancelContentCardsUpdatesSubscription(viewController);
}
}
, {
  @"Present Content Cards Info view controller", @"",
      ^(ReadmeViewController *_Nonnull viewController) {
        presentContentCardsInfoViewController(viewController);
      }
}
}
;

File: Examples/Swift/Sources/ContentCards-Custom-UI/Readme.swift
import BrazeKit
import UIKit

let readme =
  """
  This sample presents how to implement your own custom Content Cards UI:

  - AppDelegate.swift:
    - Content cards API usage
    - Content cards custom UI presentation
  - CardsInfoViewController.swift:
    - UIViewController subclass presenting the content cards data in a table view.
  """

@MainActor
let actions: [(String, String, @MainActor (ReadmeViewController) -> Void)] = [
  (
    "Print current Content Cards",
    "",
    printCurrentContentCards
  ),
  (
    "Print current Content Cards Raw",
    "",
    printCurrentContentCardsRaw
  ),
  (
    "Refresh Content Cards",
    "",
    refreshContentCards
  ),
  (
    "Subscribe to Content Cards updates",
    "",
    subscribeToContentCardsUpdates
  ),
  (
    "Cancel Content Cards updates subscription",
    "",
    cancelContentCardsUpdatesSubscription
  ),
  (
    "Present Content Cards Info view controller",
    "",
    presentContentCardsInfoViewController
  ),
]

// MARK: - Internal

@MainActor
func printCurrentContentCards(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.printCurrentContentCards()
}

@MainActor
func printCurrentContentCardsRaw(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.printCurrentContentCardsRaw()
}

@MainActor
func refreshContentCards(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.refreshContentCards()
}

@MainActor
func subscribeToContentCardsUpdates(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.subscribeToContentCardsUpdates()
}

@MainActor
func cancelContentCardsUpdatesSubscription(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.cancelContentCardsUpdatesSubscription()
}

@MainActor
func presentContentCardsInfoViewController(_ viewController: ReadmeViewController) {
  (UIApplication.shared.delegate as? AppDelegate)?.presentContentCardsInfoViewController()
}
